File: OOPSeminar5/MVCExample/MVCStudentApp/src/Model/Core/Student.java
package Model.Core;

/**
 * Дочерний класс для студента
 */
public class Student extends Person implements Comparable<Student> {
    // Поля
    private int id;
    private static int generalID;

    // Статическое поле для идентификатора студента
    static {
        generalID = 0;
    }

    /**
     * Конструктор
     * 
     * @param name // Имя
     * @param age  // Возраст
     */
    public Student(String name, int age) {
        super(name, age);
        this.id = generalID;
        generalID++;
    }

    // Методы
    public int getId() {
        return id;
    }

    // Переопределение метода toString
    @Override
    public String toString() {
        return super.toString() + String.format("; id = %s", getId());
    }

    // Переопределение метода compareTo
    @Override
    public int compareTo(Student o) {
        System.out.println(super.getName() + "-" + o.getName());
        if (super.getAge() < o.getAge())
            return -1;
        else if (super.getAge() > o.getAge())
            return 1;
        if (this.getId() < o.getId())
            return -1;
        else if (this.getId() > o.getId())
            return 1;
        return 0;
    }
}

File: OOPSeminar5/python_mvc_app/model/core/student.py
from model.core.person import Person


class Student(Person):
    generalId = 0

    def __init__(self, name: str, age: int):
        super().__init__(name, age)
        self.id = Student.generalId
        Student.generalId += 1

    def getId(self) -> int:
        return self.id

    def __str__(self) -> str:
        return f"Students [age={super().getAge()}, name={super().getName()}, id={self.id}]"